Improve test bot stability

timvandermeij opened the pull request:
Fixes #11998.
Hopefully improves #12123.

timvandermeij commented:
/botio test

timvandermeij committed:
Fix a bug in the temporary folder check in the test runner

The `noPrompt` option doesn't exist and should be `noPrompts`.

timvandermeij commented:
/botio test

timvandermeij commented:
Hm, not sure what's up with Windows here. Maybe an old process is still running?

Snuffleupagus commented on test/test.js:
```diff
@@ -828,8 +827,9 @@ function startBrowsers(rootUrl, initSessionCallback) {
       `&delay=${options.statsDelay}` +
       `&masterMode=${options.masterMode}`;
     const startUrl = rootUrl + queryParameters;
+    const headless = options.strictVerify;
```
*Probably bike-shedding, but anyway:* Would it make *slightly* more sense to use e.g. the `noPrompts` option here since that's "feels" somewhat easier to logically related to a no-UI mode!?

timvandermeij commented:
/botio test

timvandermeij committed:
Limit the allowed versions for Jasmine and Puppeteer

Jasmine >= 3.6.0 causes intermittent test failures because of random
task abortions.

Puppeteer >= 4.0.0 causes ENOTEMPTY/EBUSY errors during shutdown on the
Windows bot.

Moreover, `jasmine-core` is a dependency of `jasmine` so it doesn't have
to be required separately.

timvandermeij commented:
/botio unittest

timvandermeij commented:
/botio unittest

timvandermeij commented:
/botio test

timvandermeij commented:
The Windows bot needs a kick first before continuing here.

timvandermeij commented:
/botio test

Snuffleupagus reviewed (approved):
Looks good to me; thanks for getting the tests working again!
